src/moonlink/src/table_handler.rs

```diff
@@ -240,14 +240,11 @@ impl TableHandler {
                                 if let Err(e) = table.append(row) {
                                     warn!(error = %e, "failed to append row");
                                 }
-                                if table.should_flush() {
```

Comment:
@nbiscaro 
I think this PR change could be a regression for initial copy, because we buffer all rows in the memory rather than flushing to disk.

There're two ways to resolve I think:
- Break the current invariant, to which doesn't persist iceberg snapshot for initial copy
- Or we could use streaming write to hold initial copied rows.